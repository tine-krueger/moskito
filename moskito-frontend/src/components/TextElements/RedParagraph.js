import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

RedParagraph.propTypes = {
    children: PropTypes.string,
    textAlign: PropTypes.string
}

export default function RedParagraph({ children, textAlign}) {
    return <Eyecatcher data-testid='eyecatcher' textAlign={textAlign}>{children}</Eyecatcher>
}

const Eyecatcher = styled.p`
    color: var(--link);
    text-align: ${props => props.textAlign === 'left' ? 'left' : 'center'};
`