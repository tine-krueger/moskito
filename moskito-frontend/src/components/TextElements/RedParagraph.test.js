import { render } from '@testing-library/react'
import 'jest-styled-components'
import RedParagraph from './RedParagraph'

const mockChildren = 'This is a nice text!'
let mockTextAlign = 'center'
const component = <RedParagraph 
children={mockChildren}
textAlign={mockTextAlign}
/>

describe('RedParagraph', () => {
    it('renders correctly', () => {
        const { container } = render(component)
        expect(container).toMatchSnapshot()
    })

    it('shows the right text', () => {
        const { getByText } = render(component)
        expect(getByText('This is a nice text!')).toBeInTheDocument()
    })

    it('changes text-align', () => {
        const{ getByTestId} = render(<RedParagraph textAlign='left' children={mockChildren}/>)
        expect(getByTestId('eyecatcher')).toHaveStyle('text-align: left') 
    })

})