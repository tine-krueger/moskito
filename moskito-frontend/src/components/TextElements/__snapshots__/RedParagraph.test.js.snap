// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedParagraph renders correctly 1`] = `
.c0 {
  color: var(--link);
  text-align: center;
}

<div>
  <p
    class="c0"
    data-testid="eyecatcher"
  >
    This is a nice text!
  </p>
</div>
`;
